<a href="https://colab.research.google.com/github/gacevedob/sql-analisis-gases-invernadero/blob/main/analisis_de_emisiones.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **📊 Informe de análisis de emisiones de CO2 en la industria 🌍**

En este análisis se utilizaron datos de la huella de carbono de productos de diferentes empresas obtenidos de nature.com 🌿. La tabla 'product_emissions' 📊 analiza las emisiones de gases de efecto invernadero atribuibles a un producto en especifico medidos en CO2 (dióxido de carbono equivalente) y se analizó la PCF por producto y etapa de producción en la que se produjeron las emisiones.

Los datos utilizados en este análisis se encuentran disponibles en [nature.com](https://www.nature.com/articles/s41597-022-01178-9).

### `product_emissions`

| field                              | data_type |
|------------------------------------|-----------|
| id                                 | VARCHAR   |
| year                               | INT       |
| product_name                       | VARCHAR   |
| company                            | VARCHAR   |
| country                            | VARCHAR   |
| industry_group                     | VARCHAR   |
| weight_kg                          | NUMERIC   |
| carbon_footprint_pcf               | NUMERIC   |
| upstream_percent_total_pcf         | VARCHAR   |
| operations_percent_total_pcf       | VARCHAR   |
| downstream_percent_total_pcf       | VARCHAR   |

Con los datos disponibles se busca dar respuesta a la pregunta se buscó responder a la pregunta **¿Qué industria tiene la mayor de emisiones de gases efecto invernadero?** 🔍

💡 Se utilizó el lenguaje de programación **Python** con las librerías Pandas y **SQLite** para analizar los datos. Primero se leyeron los datos desde el archivo 'emisiones_por_industria.csv' y se creó una tabla 'emisiones_por_industria' en una base de datos SQLite. Posteriormente, se realizó una consulta SQL para obtener los datos relevantes. Se seleccionó la columna de 'industry_group' para agrupar las industrias, la columna de 'carbon_footprint_pcf' para sumar la huella de carbono de cada producto en las distintas industrias y la columna 'year' para seleccionar los datos del año 2017. La consulta se ordenó por la huella de carbono total de mayor a menor. 💻🐍

In [ ]:
import sqlite3
import pandas as pd
conn = sqlite3.connect('test.db')

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Portfolio/emisiones_por_industria.csv')
df.columns = df.columns.str.strip()
connection = sqlite3.connect('demo.db')
df.to_sql('emisiones_por_industria', connection, if_exists='replace')

In [ ]:
%load_ext sql
%sql sqlite:///demo.db

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


'Connected: @demo.db'

In [ ]:
%%sql

SELECT *
FROM emisiones_por_industria
LIMIT 5

 * sqlite:///demo.db
Done.


[(0, '10056-1-2014', 2014, 'Frosted Flakes(R) Cereal', 'Kellogg Company', 'USA', '"Food, Beverage & Tobacco"', 0.7485, 2.0, '57.50%', '30.00%', '12.50%'),
 (1, '10056-1-2015', 2015, '"Frosted Flakes, 23 oz, produced in Lancaster, PA (one carton)"', 'Kellogg Company', 'USA', 'Food & Beverage Processing', 0.7485, 2.0, '57.50%', '30.00%', '12.50%'),
 (2, '10222-1-2013', 2013, 'Office Chair', 'KNOLL INC', 'USA', 'Capital Goods', 20.68, 72.54, '80.63%', '17.36%', '2.01%'),
 (3, '10261-1-2017', 2017, 'Multifunction Printers', '"Konica Minolta, Inc."', 'Japan', 'Technology Hardware & Equipment', 110.0, 1488.0, '30.65%', '5.51%', '63.84%'),
 (4, '10261-2-2017', 2017, 'Multifunction Printers', '"Konica Minolta, Inc."', 'Japan', 'Technology Hardware & Equipment', 110.0, 1818.0, '25.08%', '4.51%', '70.41%')]

In [ ]:
%%sql

SELECT industry_group,
        COUNT(*) AS count_industry,
        ROUND(SUM(carbon_footprint_pcf),1) AS total_industry_footprint
FROM emisiones_por_industria
GROUP BY industry_group, year
HAVING year=2017
ORDER BY total_industry_footprint DESC;

 * sqlite:///demo.db
Done.


[('Materials', 11, 107129.0),
 ('Capital Goods', 4, 94942.7),
 ('Technology Hardware & Equipment', 22, 21865.1),
 ('"Food, Beverage & Tobacco"', 22, 3161.5),
 ('Commercial & Professional Services', 2, 740.6),
 ('Software & Services', 1, 690.0)]

Después de realizar la consulta en la base de datos, se encontró que la **industria de Materiales** 🏭 tuvo la **mayor cantidad de emisiones de CO2 en el año 2017** con un total de 107129.0 en carbon_footprint_pcf. Le sigue la industria de Capital Goods 🏢 con 94942.7, y Technology Hardware & Equipment 💻 con 21865.1. Las industrias de "Food, Beverage & Tobacco" 🍔, Commercial & Professional Services 💼 y Software & Services 🖥️ tuvieron menores emisiones en el año 2017.

**Es importante destacar que este análisis solo se enfocó en los datos disponibles en la tabla 'product_emissions', por lo que podrían existir otras industrias con mayores emisiones si se consideran otras fuentes de información.** 🤔